### ISSUE TITLE
Pas d'affichage quand checkbox sans label

### ISSUE THREAD

**Issue by totakoko** [codegouvfr/react-dsfr#281]
Hello ! :wave: 

Juste pour info, sur [FCU](https://github.com/betagouv/france-chaleur-urbaine), on utilise un tableau avec une colonne qui ne contient que des checkbox sans label (label='').
Depuis peu avec https://github.com/codegouvfr/react-dsfr/pull/275, l'élément label disparait mais également le style de la checkbox, car le DSFR utilise le label pour placer sa checkbox custom au dessus de l'input.

```css
.fr-checkbox-group input[type=checkbox] + label::before {
//...
}
```

=> On peut s'en sortir en mettant un espace en label. ou sinon il faudra qu'on copie le style et qu'on implémente nous-mêmes une checkbox sans label.

En l'état, ça veut dire que si pas de label, la checkbox react-dsfr ne s'affichera jamais, donc elle n'est pas vraiment utilisable. :shrug:

**Comment by garronej** [comment]
Hello @totakoko,  

Merci pour le rapport @ddecrulle, je te laisse voir ça?

**Comment by ddecrulle** [comment]
Bonjour, 

Dans ce cas je pense qu'il ne faut pas utiliser la Checkbox de la librairie. En effet elle est est conçu pour afficher plusieurs `<input type="checkbox" />`, entourés d'un `<fieldset />` etc...

Le cas que tu mentionnes a d'ailleurs été pensé par le dsfr. Tu verras qu'il y a bien un label (pour l'accessibilité) qui n'est simplement pas affiché. Tu peux retrouver ça [là](https://main--ds-gouv.netlify.app/example/component/table/) : **Tableau avec lignes sélectionnables**

**Comment by totakoko** [comment]
J'avais vu ce nouveau tableau mais il n'est pas (encore) implémenté dans la lib react-dsfr, donc c'est pas utilisable de notre côté. On a préféré utiliser la [Data Grid MUI](https://mui.com/x/api/data-grid/data-grid/) pour avoir le tri / pagination auto.
Mais donc on perd la finesse des classes DSFR.

Aussi en regardant comment ils ont fait, je suis pas méga convaincu de la règle CSS spécifique qui passe le label en transparent. Ça ne fonctionnera que pour une cellule fixed (qui est sticky), cad juste la sélection de ligne. Or chez nous, ce n'est pas une sélection de ligne mais juste un état d'email envoyé ou non.
![image](https://github.com/user-attachments/assets/5cd1b934-12f2-47a6-9ea4-b9b92501f553)

En tout cas avec un espace c'est OK pour nous, ça pourrait juste casser pour d'autres personnes.

**Comment by garronej** [comment]
@totakoko Oui, je pense que c’est une bonne idée d’utiliser Data Grid MUI.
Merci @ddecrulle pour le partage du nouveau tableau. Par contre, il est complètement headless, donc il y a beaucoup de plomberie à mettre en place pour le faire fonctionner.
Il faut, par exemple, implémenter soi-même la pagination, le tri, etc.
C’est un problème qui a déjà été résolu mille et une fois. Réimplémenter ce type de logique plutôt que de se concentrer sur la logique métier n’est pas justifié à mon avis.
Après, bien sûr, on pourrait implémenter cette logique pour vous côté react-dsfr, mais il y a de grandes chances que ce que nous produirions soit moins flexible que MUI Datagrid. Compte tenu du fait que la data grid de MUI est déjà [adaptée au DSFR automatiquement](https://github.com/codegouvfr/react-dsfr/blob/137c8a638bc70b8b0aa3181c8a02dd5d8975702f/src/mui.tsx#L226-L319), je ne sais pas si c’est très rentable d’investir du temps dans l’implémentation de ce tableau.
À mon avis, mieux vaut conseiller aux gens d’utiliser MUI pour cela.

D’ailleurs, @totakoko, utilisez-vous le MUI provider de react-dsfr ?

**Comment by totakoko** [comment]
Oui le composant DSFR est fait juste pour un tableau statique très simple.

Pour le style automatique de la Data Grid MUI, et bien je suis pas très sûr, comment je peux vérifier ça ? Car on a dû faire des [styles un peu custom](https://github.com/betagouv/france-chaleur-urbaine/blob/729d2d20ddef3dfae80f1a0b7f17222fbaca053e/src/components/ui/Table.tsx#L77-L102)...

**Comment by totakoko** [comment]
Dans ce cas oui alors je suppose qu'on a ces styles. C'est sur notre (grosse) branche [migrate_to_codegouvfr_react_dsfr](https://github.com/betagouv/france-chaleur-urbaine/tree/migrate_to_codegouvfr_react_dsfr) qu'on espère merger cette semaine...

Là on essaie de migrer de lib en restant le plus iso-fonctionnel possible, sachant que FCU contient plein d'endroits pas super carrés niveau DSFR, mais ça viendra... :)

**Comment by totakoko** [comment]
Merci à vous 2 pour votre super réactivité ! :zap: :muscle:

**Comment by totakoko** [comment]
Hm pas sûr car sans style custom on a :
![image](https://github.com/user-attachments/assets/8ef82001-ab73-4747-bf74-35c9b732bb90)

Et avec :
![image](https://github.com/user-attachments/assets/48cf5100-266e-4a99-a2af-741312b66cd9)

C'est principalement pour avoir le header différent + bordure noire au dessous.

**Comment by garronej** [comment]
@totakoko par contre je vois que vous utilisez MUI mais vous n'avez pas mis en place la configuration d'emotion pour le SSR dans Next.  

Vous pouvez suivre le guide ici: https://mui.com/material-ui/integrations/nextjs/#pages-router  

Ou encore plus simple vous pouvez appliquer la même configuration que dans le demo setup Next.js Pages Router: 
https://github.com/garronej/react-dsfr-next-demo (c'est dans pages/_app.tsx et pages/_document.tsx).  

Dans le demo setup j'utilise le tooling de tss-react: https://docs.tss-react.dev/ssr/next.js parce que le package `@mui/material-nextjs` n'existe que depuis recement mais de toute façon ils on repris l'implémentation de TSS donc c'est la même chose.

**Comment by martinratinaud** [comment]
@garronej Super merci

Pour info, ton 2e ici est plutot [ici](https://github.com/garronej/react-dsfr-next-demo/blob/49a2d1e2c881c358f3ef4b7e204de2f6617e4762/pages/_app.tsx#L42-L46) :-) (il manquait la ligne d'export)

peux tu stp m'expliquer à quoi ça sert cette config ?

Merci d'avance

**Comment by garronej** [comment]
En fait, avec Next.js, la page envoyée au client a déjà été rendue.
Vous n’envoyez pas juste un gros bundle JavaScript, vous envoyez une page déjà construite.
Or, quand vous utilisez MUI, celui-ci utilise Emotion comme moteur de style. Emotion est un moteur de style dynamique qui injecte les styles dans le head du document au moment où le composant est rendu.
Il le fait en utilisant l’API DOM. Or, quand vous faites le rendu sur le backend, le DOM n’existe pas. Il faut donc collecter les styles qui sont ajoutés et les mettre manuellement dans le `<head />`.
Si vous ne le faites pas, vos composants MUI ne sont initialement pas stylés. Vous ne l’avez peut-être pas remarqué, car dès que l’app est hydratée côté client, ces styles sont ajoutés.
Concrètement, si vous n’implémentez pas l’injection des styles en SSR, les conséquences sont que les utilisateurs ayant JavaScript désactivé reçoivent une page avec les composants MUI sans style. De plus, vous aurez un genre de scintillement avec la page qui initialement n’est pas complètement stylée pendant quelques frames.

**Comment by martinratinaud** [comment]
Super explication merci

C'est implémenté

**Comment by garronej** [comment]
J'ai pas précisé mais vous devez ajouter `@emotion/server et tss-react` a vos dépendences

**Comment by garronej** [comment]
Mais honètement c'est un peut relou qu'Emotion n'exporte pas directement son propre utilitaire. J'ai essayer de pousser pour qu'ils le fasse mais ça n'a pas pris. J'ai au moins réussi a convraincre MUI de le faire (enfin j'ai été parmis ceux plutot)
